Testnet node release 12 Jan 2023 (#712)
Concordium/concordium.github.io: source/mainnet/net/resources/release-notes.rst

   To subscribe to updates on the Testnet status page click **Subscribe** to get all updates or click **Get updates** to choose to get all updates or only updates for specific products.

January 12, 2023

Concordium Node 5.1.3
---------------------

Concordium node version 5.1.3 introduces the following new features and improvements:

> - Improvements were made to allow greater concurrency with transaction processing. (Checking transaction signatures is done without acquiring the global state lock.)

Reviewer: ```suggestion
- Improvements were made to allow greater concurrency with transaction processing.
```
Some other checks are also done without the lock now. I would not go into those details.

> - Blocks are relayed earlier. In particular this means that blocks are now processed in two steps: block receive and block execute. Blcok receive performs verification of block metadata while the block execute adds the block to the tree. Blocks are now enqueued in the outgoing message queue in between the two steps.

Reviewer: ```suggestion
- Blocks are relayed earlier. This decreases the time it takes for the network to become aware of a block.
```

> - Removed the configuration option ``no_rebroadcast_consensus_validation``.

Reviewer: ```suggestion
- Removed the configuration option ``no_rebroadcast_consensus_validation``. This option (which was used for testing only and was disabled by default) made the node rebroadcast blocks before doing any validation.
```

- Changes were made to avoid deadlocks during node shutdown in specific scenarios.

- The node will now shut down to start if an error occurs in a required service (e.g., grpc server). In particular, the node will shut down if a required service could not be started.

- Added timeout to downloading out of band catchup files when block indices and catch-up chunk files are specified by an URL. The timeout is controlled by the option ``--download-blocks-timeout`` (environment variable ``CONCORDIUM_NODE_CONSENSUS_DOWNLOAD_BLOCKS_TIMEOUT``) and defaults to 5 minutes. Timeout is now five minutes per chunk instead of waiting indefinitely.

> - Removed the "instrumentation" feature of the node and built the node with Prometheus support enabled by default.
> 

Reviewer: ```suggestion
```
This was an internal thing so I would not include it in this changelog.

Reviewer: It is a build-time configuration, it is not a feature that was ever enabled for any of the distributions builds.
